pkg/api/upload.go

```diff
@@ -0,0 +1,221 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/virtual-vgo/vvgo/pkg/projects"
+	"github.com/virtual-vgo/vvgo/pkg/sheets"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type UploadType string
+
+const (
+	UploadTypeClix   UploadType = "clix"
+	UploadTypeSheets UploadType = "sheets"
+)
+
+type UploadHandler struct {
+	sheets.Sheets
+}
+
+type Upload struct {
+	UploadType    `json:"upload_type"`
+	*ClixUpload   `json:"clix_upload"`
+	*SheetsUpload `json:"sheets_upload"`
+	Project       string `json:"project"`
+	FileName      string `json:"file_name"`
+	FileBytes     []byte `json:"file_bytes"`
```

Comment:
I still have concerns over this. What happens when we need to go multipart, say, when someone uploads 4k video? Base64 in JSON can't get chunked.

Reply:
You can chunk base64 encoding. You just only encode the x bytes of the pdf, then the next x bytes and so on. Altho, this api doesn't address multi-part uploads just yet.

Reply:
That seems mildly horrifying, compared to just separating metadata from the actual file.

Reply:
I'm not sure what you mean?

Reply:
For partial uploads, we'll need to track the parts and put them back together in the correct order, but I don't think that will be too hard.

Reply:
Still, base64 imposes a 33% overhead.